src/register.rs

```diff
@@ -20,13 +20,14 @@ pub struct Register<const N: usize> {
     /// The state vector is [a, b, c, ...]<sup>T</sup>, where |state_i|<sup>2</sup> represents the probability
     /// that the system will collapse into the state described by the ith basis vector.
     pub state: Array2<Complex<f64>>, // Should not be pub (it is pub now for testing purpouses)
+    size: usize
 }
 
 
-impl<const N: usize> Register<N> {
+impl Register {
 
     /// Creates a new state with an array of booleans with size N 
-    pub fn new(input_bits: [bool; N]) -> Self {
+    pub fn new(input_bits: Vec<bool>) -> Self {
```

Comment:
This could be replaced with a slice `&[bool]`

Reply:
Should be changed now in the latest commit